### PR TITLE
chore: add validation to detect pipe characters in asset paths

### LINKED ISSUE #338: Feature request: Prevent or warn customers from submitting jobs with "|" in the file or path name
### Describe the problem

Currently, if there is "|" in the path or file name we fail to sync the asset on Windows workers with the following error. 

If there is "|" in asset file name. 
```
2025/11/08 16:57:20-08:00 [Errno 22] Invalid argument: 'C:\\Sessions\\session-abc\\assetroot-abc\\tex\\Map-COL-with"|".jpg.af4d5BFa'
```

If there is "|" in path name. 

```
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\Sessions\\session-abc\\assetroot-abc\\tex with | '
```

### Proposed Solution

Instead of uploading the assets and failing when downloading the assets we should be preventing customers from submitting these scene or at least let them know that Windows doesn't allow "|" character. 

### Example Use Cases

Whenever someone submits a job with "|" in the file name or path name.

### PR DESCRIPTION
**Pull request by vaish-gujjari**
Fixes: https://github.com/aws-deadline/deadline-cloud-for-cinema-4d/issues/338

### What was the problem/requirement? (What/Why)
Currently, if there is "|" in the path or file name we fail to sync the asset on Windows workers with the following error.

If there is "|" in asset file name.

2025/11/08 16:57:20-08:00 [Errno 22] Invalid argument: 'C:\\Sessions\\session-abc\\assetroot-abc\\tex\\Map-COL-with"|".jpg.af4d5BFa'
If there is "|" in path name.

OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\Sessions\\session-abc\\assetroot-abc\\tex with | '

### What was the solution? (How)
Created a path_validator module with a validate_asset_paths() function that iterates through all asset paths and checks for the pipe character "|". When detected, it logs a warning through the WarningCollectorHandler which displays the message in the submitter UI, alerting users before job submission that these paths will cause sync failures on Windows workers.

Screenshot 2025-11-20 at 2 50 43 PM

### What is the impact of this change?
Users will now receive clear warnings in the submitter UI when their asset paths contain pipe characters, allowing them to fix the issue before submitting jobs. This prevents job failures on Windows workers and saves time by catching the problem early rather than discovering it after submission when assets fail to sync.

### How was this change tested?
Added comprehensive unit tests in test_path_validator.py covering:
- Detection of pipe characters in filenames
- Detection of pipe characters in directory names
- Handling multiple paths with pipe characters
- Validation that clean paths don't trigger warnings
- Edge case of empty asset sets
All tests pass along with linting (ruff, black, mypy) validation.

- Have you made changes to the submitter?
No

### Was this change documented?
No

### Is this a breaking change?
No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/deadline/cinema4d_adaptor/Cinema4DClient/cinema4d_client.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def graceful_shutdown(self, signum: int, frame: FrameType | None):
         sys.exit(0)
 
     def map_path(self, path: str) -> str:
-        """
+        r"""
```
Comment (karthikbekalp) [inline src/deadline/cinema4d_adaptor/Cinema4DClient/cinema4d_client.py]: Typo?

File: src/deadline/cinema4d_submitter/assets.py [hunk 2]
```diff
@@ -26,6 +27,9 @@ def parse_scene_assets(self) -> set[Path]:
         # Clear any previous warnings before processing
         warning_collector.clear_warnings()
 
+        # Clear any previous warnings before processing
+        warning_collector.clear_warnings()
```
Comment (karthikbekalp) [inline src/deadline/cinema4d_submitter/assets.py]: Any reason we have a duplicate?

**Review by karthikbekalp — CHANGES_REQUESTED** [review]
I think the changes look good, but can you also add images on how the feature will look like?

File: src/deadline/cinema4d_adaptor/Cinema4DClient/cinema4d_client.py (hunk 1 above)
Comment (vaish-gujjari) [reply src/deadline/cinema4d_adaptor/Cinema4DClient/cinema4d_client.py]: Hi Karthik, no. So when we run "hatch run fmt" we're getting the below issue:

test/unit/deadline_submitter_for_cinema4d/test_warning_collector.py::TestWarningCollector::test_get_warnings_returns_copy <unknown>:47: SyntaxWarning: invalid escape sequence '\p'
~\Desktop\deadline-cloud-for-cinema-4d\src\deadline\cinema4d_adaptor\Cinema4DClient\cinema4d_client.py:47: SyntaxWarning: invalid escape sequence '\p'
"""
<unknown>:47: SyntaxWarning: invalid escape sequence '\p'
~\Desktop\deadline-cloud-for-cinema-4d\src\deadline\cinema4d_adaptor\Cinema4DClient\cinema4d_client.py:47: SyntaxWarning: invalid escape sequence '\p'
"""
<unknown>:47: SyntaxWarning: invalid escape sequence '\p'
<unknown>:47: SyntaxWarning: invalid escape sequence '\p'
~\Desktop\deadline-cloud-for-cinema-4d\src\deadline\cinema4d_adaptor\Cinema4DClient\cinema4d_client.py:47: SyntaxWarning: invalid escape sequence '\p'
"""

By adding r before """,  we're treating all the characters as literals and not interpret as escape sequences..

File: src/deadline/cinema4d_submitter/assets.py (hunk 2 above)
Comment (vaish-gujjari) [reply src/deadline/cinema4d_submitter/assets.py]: Good catch! This was an oversight on my part. I've cleaned it up.

File: src/deadline/cinema4d_adaptor/Cinema4DClient/cinema4d_client.py (hunk 1 above)
Comment (karthikbekalp) [reply src/deadline/cinema4d_adaptor/Cinema4DClient/cinema4d_client.py]: Hmm, I don't think adding "r" is the right solution here because: 

1. The code currently in mainline passes the formatter test (i.e. `hatch run fmt` is successful)

2. I think we want to deep dive why we are getting this error in the first place. The error is raised by "test_get_warnings_returns_copy" which is a submitter code trying to check client code which does not look right. Maybe there's an issue in your workspace? 

```
test/unit/deadline_submitter_for_cinema4d/test_warning_collector.py::TestWarningCollector::test_get_warnings_returns_copy :
```

3. The ":47: SyntaxWarning: invalid escape sequence '\p'" is interesting because it does bring up the issue that we do have values in the docstrings that could be misinterpreted as escape characters. 

Let's try to debug this offline once you head over... :)

**Review by karthikbekalp — CHANGES_REQUESTED** [review]
